### PR TITLE
feat(python): support src/packagename/__init__.py

### PR DESCRIPTION
**Pull request by scop**
For the popular src directory layout,
https://blog.ionelmc.ro/2014/05/25/python-packaging/#the-structure

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/release-please/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Refs #1026 

Basically just monkeying what was done with earlier related python changes recently and git grepping what might need updating without really understanding it all, could very well be that I've missed something.

### CODE REVIEW

File: src/releasers/python.ts [hunk 1]
```diff
@@ -115,6 +115,14 @@ export class Python extends ReleasePR {
         packageName: packageName.name,
       })
     );
+    updates.push(
```
Comment (bcoe) [inline src/releasers/python.ts:118]: Mind adding a `// TODO: figure out refactor that makes logic for updating __init__.py, etc., configurable`

I'm a little worried about the number of optional paths we've added at this point.

**Review by bcoe — CHANGES_REQUESTED** [review]

File: src/releasers/python.ts (hunk 1 above)
Comment (scop) [reply src/releasers/python.ts:119]: Added.

FWIW after this one's in, pretty much the last one re filenames I'm aware some projects could appreciate would be updating `__version__.py` files anywhere in the tree. That could be handled just like `version.py` currently is.

But at that point I think all the "standard" ways to refer to the version would be addressed and a reasonable alternative to configurability might be to nudge projects to use a more common filename to place their `__version__` in.

Then again while the `__version__` _symbol_ is probably the most prominent, some projects use just `version` or `VERSION`...

**Review by bcoe — APPROVED** [review]
